### PR TITLE
BitVector constructor from NTuple

### PR DESCRIPTION
**Pull request by matbesancon**
Avoids allocating a vector to build BitVector

### CODE REVIEW

**Comment by mbauman** [comment]
I think this should be done much more consistently and thoroughly, as part of #24595.  Now that we have `undef` initializers, I think we could support _any_ iterable for _all_ `AbstractArray` constructors.

**Comment by matbesancon** [comment]
the advantage of `NTuple` instead of `iter::Any` is that one can leverage the size and constrain the type of input

File: base/bitarray.jl [hunk 1]
```diff
@@ -71,6 +71,28 @@ const BitMatrix = BitArray{2}
...
+
+julia> BitVector(nt)
+4-element BitArray{1}:
+ 1
+ 0
+ 1
+ 0
+```
+"""
+function BitVector(nt::NTuple{N, Bool}) where {N}
```
Comment (JeffBezanson) [inline base/bitarray.jl]: This should be `Tuple{Vararg{Bool}}` to avoid forcing us to specialize it on every length.

Comment (matbesancon) [reply base/bitarray.jl]: Done thanks

**Comment by StefanKarpinski** [comment]
Bump...

**Comment by matbesancon** [comment]
do we want to restart CI for the failing package_macos build bot?

**Comment by StefanKarpinski** [comment]
🤷‍♂ It's a known issue and clearly unrelated. Seems unlikely that this change would fail just on Mac.

**Pull request merged**